### ISSUE TITLE
[Feature] Add support for newly announced A1

### ISSUE THREAD

**Issue by AdrianGarside** [greghesp/ha-bambulab#303]
### Describe the feature

Need log output that includes the model version data.

### What device is this for?

A1 Mini

**Comment by AdrianGarside** [comment]
#304

**Comment by bjeanes** [comment]
FWIW I was able to add my new A1 as A1 Mini and afaict all sensors are present and make sense.

Is there any debug info I can fetch to help confirm that it works or to finalise official support?

**Comment by AdrianGarside** [comment]
Yes! You can grab the logs as described at the bottom of the readme:
https://github.com/greghesp/ha-bambulab/blob/main/README.md

And you can then try out the new version which should have my 'best guess' at A1 support (mostly what the codename for the controller is) and see if it just works:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0-dev7

You'll need to set the checkbox to show beta versions to be able to install that build.

**Comment by bjeanes** [comment]
Here's diagnostics: [config_entry-bambu_lab-c9b0fc8146a9e1fe7dec7e82e453e497.json-3.txt](https://github.com/greghesp/ha-bambulab/files/13747825/config_entry-bambu_lab-c9b0fc8146a9e1fe7dec7e82e453e497.json-3.txt)


Logs don't contain much, even with debug on (and pressing manually update). Just lots of repeated versions of:

```
2023-12-22 14:23:55.379 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab] Manually updated bambu_lab data
2023-12-22 14:23:55.500 DEBUG (SyncWorker_39) [custom_components.bambu_lab] P1PImage::async_image
2023-12-22 14:23:56.134 DEBUG (Thread-21759) [custom_components.bambu_lab.pybambu] JPEG received
2023-12-22 14:23:56.165 DEBUG (Thread-21759) [custom_components.bambu_lab.pybambu] JPEG received
2023-12-22 14:23:56.193 DEBUG (Thread-21759) [custom_components.bambu_lab.pybambu] JPEG received
2023-12-22 14:23:57.221 DEBUG (Thread-21759) [custom_components.bambu_lab.pybambu] JPEG received
2023-12-22 14:23:57.253 DEBUG (Thread-21759) [custom_components.bambu_lab.pybambu] JPEG received
2023-12-22 14:23:57.391 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab.pybambu] Message: A1Mini: b'{"print":{"nozzle_temper":48.34375,"bed_temper":40.9375,"command":"push_status","msg":1,"sequence_id":"57444"}}'
```

**Comment by bjeanes** [comment]
> The new version supports configuration via Bambu Cloud and should just show your A1 as an available printer to choose.

Unfortunately this seems to fail:

<img width="436" alt="Screenshot 2023-12-22 at 2 36 59 pm" src="https://github.com/greghesp/ha-bambulab/assets/2560/fb75cc9c-3993-4974-8dca-146025649561">

However, manually configuring and choosing A1 seems to work, other than two "Unavailable" entities showing up ("Active tray" and "Active tray index"):

<img width="702" alt="Screenshot 2023-12-22 at 2 42 34 pm" src="https://github.com/greghesp/ha-bambulab/assets/2560/7bc27bf3-eeea-4285-b575-607a21821c99">


<img width="1050" alt="Screenshot 2023-12-22 at 2 41 36 pm" src="https://github.com/greghesp/ha-bambulab/assets/2560/a440911d-347e-4b62-a81e-6915c184f2a8">

**Comment by bjeanes** [comment]
The only issue that I've noticed is that the entity IDs still started with `a1mini_<serial>`. The AMS and External Spool devices OTOH were correctly prefixed as `a1_<serial>`. I manually renamed my device anyway, so it doesn't really matter.

**Comment by bjeanes** [comment]
> Logs don't contain much, even with debug on

I take that back, my new computer meant that I didn't have a large terminal scroll back configured and missed what you're probably looking for:

```
2023-12-22 14:05:36.016 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab] Manually updated bambu_lab data
2023-12-22 14:05:36.513 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Sent {'pushing': {'sequence_id': '0', 'command': 'pushall'}} to topic device/REDACTED/request
2023-12-22 14:05:36.514 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Sent {'info': {'sequence_id': '0', 'command': 'get_version'}} to topic device/REDACTED/request
2023-12-22 14:05:36.571 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab.pybambu] Message: A1Mini: b'{"info":{"command":"get_version","sequence_id":"0","module":[{"name":"ota","project_name":"N2S","sw_ver":"01.01.00.01","hw_ver":"OTA","sn":"REDACTED"},{"name":"esp32","project_name":"N2S","sw_ver":"01.07.22.25","hw_ver":"AP05","sn":"REDACTED"},{"name":"mc","project_name":"N2S","sw_ver":"00.00.20.62","loader_ver":"00.00.00.32","hw_ver":"MC02","sn":"REDACTED"},{"name":"th","project_name":"N2S","sw_ver":"00.00.06.74","loader_ver":"00.00.00.26","hw_ver":"TH01","sn":"REDACTED"},{"name":"ams_f1/0","project_name":"","sw_ver":"00.00.07.89","loader_ver":"00.00.00.00","ota_ver":"00.00.00.00","hw_ver":"AMS_F102","sn":"REDACTED"}],"result":"success","reason":""}}'
2023-12-22 14:05:36.572 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab.pybambu] Got Version Data
```

**Comment by AdrianGarside** [comment]
> The only issue that I've noticed is that the entity IDs still started with `a1mini_<serial>`. The AMS and External Spool devices OTOH were correctly prefixed as `a1_<serial>`. I manually renamed my device anyway, so it doesn't really matter.

That will probably be because you initially set up as an A1 Mini and home assistant remembers the earlier IDs when handed a sensor with the same underlying unique serial id.

**Comment by AdrianGarside** [comment]
“Unfortunately this seems to fail:” - anything in the debug logs around this failure? 2FA isn’t supported.

**Comment by AdrianGarside** [comment]
> > Logs don't contain much, even with debug on
> 
> I take that back, my new computer meant that I didn't have a large terminal scroll back configured and missed what you're probably looking for:
> 
> ```
> 2023-12-22 14:05:36.016 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab] Manually updated bambu_lab data
> 2023-12-22 14:05:36.513 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Sent {'pushing': {'sequence_id': '0', 'command': 'pushall'}} to topic device/REDACTED/request
> 2023-12-22 14:05:36.514 DEBUG (MainThread) [custom_components.bambu_lab.pybambu] Sent {'info': {'sequence_id': '0', 'command': 'get_version'}} to topic device/REDACTED/request
> 2023-12-22 14:05:36.571 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab.pybambu] Message: A1Mini: b'{"info":{"command":"get_version","sequence_id":"0","module":[{"name":"ota","project_name":"N2S","sw_ver":"01.01.00.01","hw_ver":"OTA","sn":"REDACTED"},{"name":"esp32","project_name":"N2S","sw_ver":"01.07.22.25","hw_ver":"AP05","sn":"REDACTED"},{"name":"mc","project_name":"N2S","sw_ver":"00.00.20.62","loader_ver":"00.00.00.32","hw_ver":"MC02","sn":"REDACTED"},{"name":"th","project_name":"N2S","sw_ver":"00.00.06.74","loader_ver":"00.00.00.26","hw_ver":"TH01","sn":"REDACTED"},{"name":"ams_f1/0","project_name":"","sw_ver":"00.00.07.89","loader_ver":"00.00.00.00","ota_ver":"00.00.00.00","hw_ver":"AMS_F102","sn":"REDACTED"}],"result":"success","reason":""}}'
> 2023-12-22 14:05:36.572 DEBUG (Thread-347 (mqtt_listen_thread)) [custom_components.bambu_lab.pybambu] Got Version Data
> ```

That says A1Mini which is incorrect. Looks like I need to make a fix once I’m back at my computer.

**Comment by AdrianGarside** [comment]
New release with some tweaks to the A1 support given your logs: https://github.com/greghesp/ha-bambulab/releases/tag/v2.0-dev9

**Comment by Mangomc** [comment]
I´m trying to setup the integration with my new A1 combo
But after the login i get an error. "no_printers"
Any way i can help to fix the issue?

**Comment by AdrianGarside** [comment]
> I´m trying to setup the integration with my new A1 combo But after the login i get an error. "no_printers" Any way i can help to fix the issue? Using 2.0dev9

Had you already added it to home assistant as a different printer. No printers should only happen if it's already been added to home assistant (identified by serial so if you lied for testing that still counts).

**Comment by Mangomc** [comment]
That Was it. Thanks. I tried the stable before but abort after the login. Now my A1 is added

**Comment by AdrianGarside** [comment]
> That Was it. Thanks. I tried the stable before but abort after the login. Now my A1 is added

Everything look correct?

**Comment by Mangomc** [comment]
Yes. Ive got a HMS error but thats something i need to find out myself. All sensors are working. Nothing unavailable.
Cam is also working. Looks perfect :)

**Comment by bjeanes** [comment]
> “Unfortunately this seems to fail:” - anything in the debug logs around this failure? 2FA isn’t supported.

I don't have 2FA enabled:

<img width="1020" alt="Screenshot 2023-12-23 at 10 23 21 am" src="https://github.com/greghesp/ha-bambulab/assets/2560/04e2f934-c731-477c-857f-d579f3c7064a">

I'll update to your latest, delete all my integrations, and re-add, checking the logs!

**Comment by bjeanes** [comment]
Huh after updating to the latest, the cloud-based login works. Maybe I had a temporary blip or the fact that I still had the previous A1Mini integration (from stable release) at the time caused an issue. From a clean slate, it seems to work perfectly!

> Everything look correct?

Does for me!

**Comment by AdrianGarside** [comment]
If you installed any of the -coverimage releases I broke the cloud login in those as I made changes to make the newly structured login data more available to be able to download the cover images. Maybe you installed one of those more alpha builds.

**Comment by bjeanes** [comment]
Ah yeah the first beta tag I used did indeed have `-coverimage` in its name, so that probably was it.

Anyway, based on my experience, I think it's safe to say A1 support is handled once you cut your next release :)

**Comment by AdrianGarside** [comment]
Fixed in v2.0.0.
